amespahdbpythonsuite/mcfitted.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env python4
+
+from __future__ import annotations
+from typing import Optional
+
+import numpy as np
+import matplotlib.pyplot as plt  # type: ignore
+
+from scipy import stats  # type: ignore
+from specutils import Spectrum1D  # type: ignore
+
+from amespahdbpythonsuite.amespahdb import AmesPAHdb
+
+message = AmesPAHdb.message
+
+
+class MCFitted:
+    """
+    AmesPAHdbPythonSuite mcfitted class.
+    Contains methods to fit and plot the input spectrum using a Monte Carlo approach.
+
+    """
+
+    def __init__(self, d: Optional[dict] = None, **keywords) -> None:
+        self.__set(d, **keywords)
+
+    def __set(self, d: Optional[dict] = None, **keywords) -> None:
+        """
+        Populate data dictionary helper.
+
+        """
+        self.mcfits = keywords.get("mcfits", list())
+        self.distribution = keywords.get("distribution", "")
+
+        if isinstance(d, dict):
+            if d.get("type", "") == self.__class__.__name__:
+                if "mcfits" not in keywords:
+                    self.mcfits = d["mcfits"]
+                if "distribution" not in keywords:
+                    self.distribuion = d["distribuion"]
+
+    def get(self) -> dict:
+        """
+        Calls class: :class:`amespahdbpythonsuite.transitions.Transitions.get`.
+        Assigns class variables from inherited dictionary.
+
+        """
+        d = {}
+        d["type"] = self.__class__.__name__
+        d["mcfits"] = self.mcfits
+        d["distribution"] = self.distribution
+
+        return d
+
+    def getstats(self, d=dict()) -> dict:
+        """
+        Get statistics for the mcfitted spectra.
+
+        Returns:
+            stat : dictionary
+
+        """
+        stat = {'mean': stats.describe(d).mean, 'std': np.sqrt(stats.describe(d).variance),
+                'skew': stats.describe(d).skewness, 'kurt': stats.describe(d).kurtosis}
+
+        return stat
+
+    def getfit(self, **keywords) -> dict:
+        """
+        Retrieves the mean, std, skewness, and curtosis spectra.
+
+        """
+        mcfits = list()
+        for mcfit in self.mcfits:
+            mcfits.append(mcfit.getfit())
+
+        return self.getstats(mcfits)
+
+    def getbreakdown(self, **keywords) -> dict:
+        """
+        Retrieves the breakdown of the MC fitted PAHs.
+
+        """
+        keys = ['solo', 'duo', 'trio', 'quartet', 'quintet',
+                'neutral', 'cation', 'anion',
+                'small', 'large',
+                'pure', 'nitrogen',
+                'nc', 'error']
+
+        results: dict = {key: [] for key in keys}
+
+        # Obtain fit breakdown.
+        for fit in self.mcfits:
+            bd = fit.getbreakdown()
+            for key in keys[:-1]:
+                results[key].append(bd[key])
+            # Obtain fit uncertainty.
+            error = fit.geterror()['err']
```

Comment:
No need to add geterror()-result, as it is already available in as a separate method.